cli/ballerina-packerina/src/test/java/org/ballerinalang/packerina/cmd/BuildCommandTest.java

```diff
@@ -717,6 +717,24 @@ public void testTargetCacheDirectory() throws IOException {
         // check if each module has a bit in cache directory
     }
 
+    @Test(description = "Test the cleaning of target resources in the build command.",
+            dependsOnMethods = {"testBuildCommand"})
+    public void testTargetClean() {
+        // If a single module is built only the relevant module's resources should be cleaned,
+        // else the entire target will be deleted.
+        String[] compileArgs = {"mymodule", "--skip-tests"};
+        Path target = this.testResources.resolve("valid-project").resolve(ProjectDirConstants.TARGET_DIR_NAME);
+
+        BuildCommand buildCommand = new BuildCommand(this.testResources.resolve("valid-project"),
+                printStream, printStream, false, true);
+        new CommandLine(buildCommand).parse(compileArgs);
+        buildCommand.execute();
+
+        // Executable of a module that is not built.
+        Path executablePath = target.resolve(ProjectDirConstants.BIN_DIR_NAME).resolve("mytemplate.jar");
```

Comment:
Where we build the mytemplate module. Another test?

Reply:
Also shall we check whether the target directory stuff cleaned ( ex. balo, jar etc)

Reply:
Yes, the `mytemplate` module is built in the dependent test `testBuildCommand`.

There isn't actually a way to check if the relevant target directory resources are cleaned in the test case since it is an intermediate step and the artifacts will get replaced. However, this was verified when debugging the code.